hydra-node/src/Hydra/Chain/Direct/Tx.hs

```diff
@@ -266,7 +315,35 @@ commitTx networkId scriptRegistry headId party utxoToCommitWitnessed (initialInp
   commitDatum =
     mkTxOutDatumInline $ mkCommitDatum party utxoToCommit (headIdToCurrencySymbol headId)
 
-  utxoToCommit = fst <$> utxoToCommitWitnessed
+  TxMetadata metadataMap = mkHydraHeadV1TxName "CommitTx"
+
+  txAuxMetadata = mkAlonzoTxAuxData @[] @LedgerEra (toShelleyMetadata metadataMap) [toLedgerScript initialScript]
+
+-- Re-Associate the redeemer indices present in some tx with the same inputs but different
+-- indices in the current tx.
+--
+-- We need this in order to be able to find the the _old_ tx redeemer and re-add
+-- it to the tx witnesses using the appropriate index of the _new_ tx input
+-- because of the 'TxIn' re-ordering.
+reAssociateOldInputs ::
+  Set (Ledger.TxIn StandardCrypto) ->
+  Set TxIn ->
+  Map (AlonzoPlutusPurpose AsIndex era) a ->
+  [(Ledger.TxIn StandardCrypto, Int, Maybe a)]
+reAssociateOldInputs newInputs oldInputs oldTxRedeemers =
```

Comment:
this function seems to be used once, why extracted it?

i think this is related to my comment above about having a separate function to resolve the ledgerBlueprintTx wits.